docs/fsharp/language-reference/type-extensions.md

```diff
@@ -95,13 +95,16 @@ Optional extensions do not appear on the extended type when examined by reflecti
 
 Optional extension members are compiled to static members for which the object instance is passed implicitly as the first parameter. However, they act as if they were instance members or static members according to how they are declared.
 
-It is not possible to define an optional extension on a generic type when the type variable is constrained. For example, the following is not possible:
+## Generic limitation of intrinsic and optional type extensions
+
+It is possible to declare a type extension on a generic type where the type variable is constrained. The requirement is that the constraint of the extension declaration matches the constraint of the declared type.
```

Comment:
Yes that's what I wanted to add, what you could add though is that specifically SRTP parameters (constrained or not) are not allowed. (I just tested this, ` ^T` already fails)